manifests/database/postgresql.pp

```diff
@@ -26,33 +32,39 @@
     # Temporary direct use of liquibase to initiall migrate the candlepin database
     # until support is added in cpdb - https://bugzilla.redhat.com/show_bug.cgi?id=1044574
     include ::postgresql::client, ::postgresql::server
-    postgresql::server::db { $candlepin::db_name:
-      user     => $candlepin::db_user,
-      password => postgresql_password($candlepin::db_user, $candlepin::db_password),
+    postgresql::server::db { $db_name:
+      user     => $db_user,
+      password => postgresql_password($db_user, $db_password),
       encoding => 'utf8',
       locale   => 'en_US.utf8',
-    } ~>
+    }
+    Postgresql::Server::Role[$db_user] -> Postgresql::Server::Database[$db_name]
```

Comment:
While you're cleaning up: this is fixes in puppetlabs-postgresql 4.2.0 so I'd prefer to remove this and increase the minimum version in metadata.json.